### DISCUSSION TITLE
Infrahub for ISP Providers

### DISCUSSION THREAD

**Discussion by ajackson79** [opsmill/infrahub#4706] [Q&A] [answered]
Does Infrahub have the data model flexibility to support being a source of truth for an ISP network? Some of the requirements that are lacking in many of the existing options are advanced routing protocol support, MPLS/SR support, Service mappings for things like L2VPN, Traffic Engineering, Service multiplexing, Complex Vlan configurations, translations, and encapsulations for S-Vlan and C-Vlan use cases.

The final question is does Infrahub have any graphical displays for things like racks, cables, etc like Netbox and Nautobot currently have available?

**Comment by BaptisteGi** [comment]
Hello,

The short answer is yes! Infrahub is built with flexibility in mind, allowing you to design your own schema to perfectly align with your requirements (advanced routing, QinQ, L2VPN ...). Follow [this guide](https://docs.infrahub.app/topics/schema) to learn more about schema.

Currently, we don’t yet support graphical displays of rack elevations or cable traces, but this feature is on our roadmap for future development. If you’d like to stay updated or discuss further, feel free to join [our Discord server](https://discord.gg/opsmill)!

Baptiste

**Reply by ajackson79** [reply]
Do you publish this roadmap anywhere?

Also since there is no cable trace option, yet, if we want to document interface connections we would have to leave out FDPs as parts of the connection for Infrahub to model and show that connection?

**Reply by BaptisteGi** [reply]
We need to publish the roadmap, but the reality is that we’re not quite ready yet

I’m not sure I fully understand your question here, did you mean SFP? At the moment the feature gap is really on the visualization aspect. You still can capture cabling related data (cables, interfaces, SFP, circuit ...), we have a schema example [in this repository](https://github.com/opsmill/infrahub-demo-dc-fabric/tree/main/models).

**Reply by ajackson79** [reply]
I was more referring to pass through devices like FDPs. Cable traces show you the physical cables, but interface connections show you which interfaces are connected to each other for things like subnets, Vlans, etc.

**Reply by BaptisteGi** [reply]
By FDP do you mean Fiber Distribution Panel? I think in your schema you could capture the physical layer (Patch panel, cables and how they connect interfaces together). Then, at interface level, capture layer 2 (e.g. tagged vlans, dot1q mode, qinq ...) or layer 3 (e.g. ip assigned ...) information. Finally it's also possible to capture routing details (ospf, static routes ...) in dedicated objects.

**Reply by ajackson79** [reply]
Yeah Fiber distribution panels. I'm looking at this from a Netbox/Nautobot stand-point. In those products FDPs are considered pass through devices, but that is very dependent on how those products work and are designed.

Being able to capture things at the various levels would be ideal. That would allow us to capture the physical layer, the DWDM layer, the switching/Vlan layer, routing layer, and finally the MPLS services layer.

Do you have any schema examples with that many layers for a single network?

**Reply by BaptisteGi** [reply, marked as answer]
We indeed have schema for similar objects! We will soon (in the upcoming days) publish schema examples, I keep you posted ;)

**Reply by BaptisteGi** [reply]
@ajackson79 here it is: https://github.com/opsmill/schema-library/tree/main/extensions/patch_panel

The idea here is to create a "patch panel" so an actual equipment you can insert into a rack. Then your patch panel has X interfaces on the front side that you could map with Y interfaces on the rear side. You can then plug cables to these interfaces to connect to a regular network device interface for instance (router, switch ...). This allows you to document the physical layer of your network.

**Reply by ajackson79** [reply]
That is perfect. I assume cable traces would then go from device interface to device interface, but include the patch panels, patch panel interfaces, etc in the trace?

**Reply by BaptisteGi** [reply]
Yes, I believe the "physical" trace should follow cables across device interface, patch panel interface front & rear ...
